cql3/statements/alter_keyspace_statement.cc

```diff
@@ -206,6 +206,14 @@ cql3::statements::alter_keyspace_statement::prepare_schema_mutations(query_proce
         auto ts = mc.write_timestamp();
         auto global_request_id = mc.new_group0_state_id();
 
+        // #22688 - filter out any dc*:0 entries - consider these
+        // null and void (removed). Migration planning will treat it
+        // as dc*=0 still.
+        std::erase_if(ks_options, [](const auto& i) {
+            auto& [key, val] = i;
+            return key.starts_with(ks_prop_defs::KW_REPLICATION) && val == "0";
```

Comment:
This seems wrong.
I'd expect a dc to not start with "replication".
Though it might.
Shouldn't filter using exact mismatch to the reserved "replication_factor" key?

Reply:
It is the flattened options: All key+value pairs are prefixed with their category, thus a dc rf will be like `"replication:dc0" : "<number>"`. For replication, the only possible options are class, replication_factor and per-dc rf. The two first cannot be zero (already checked), so thus I felt this short check should be ok. Can verify explicitly that it is neither of those opts of course...

Reply:
I see.  Sine this is not on the hot path, maybe defensively call `key.starts_with(sstring(ks_prop_defs::KW_REPLICATION) + ":")` then, in case we ever add another map starting with "replication". And add comment about the flattened options for future readers.
Can also document that `class` and `replication_factor` cannot contain `"0"`.